debian/templates

```diff
@@ -0,0 +1,11 @@
+Template: raspberrypi-sys-mods/allow_3p_repos
+Type: select
+Choices: yes, no
+Default: yes
```

Comment:
as you already stated in your comment https://github.com/RPi-Distro/raspberrypi-sys-mods/pull/51#issuecomment-774732103 the default is a `yes`. personally i think this should be a `no`.

i am not sure but i assume that running `apt-get --yes install ...` the `--yes` will also answer this question with `yes` will it? (haven't tested). if that is true the default should be `no`. 

running apt with `--yes` is often done to automated bootstrap systems, like for docker images, for use with puppet and so on.
from GDPR view the default also should not (must not?) be `yes` as the ip of the users computer is sent to microsoft without knowing. an ip address counts as personal information. tracking must be opt in and not opt out. the yes from install the package `raspberrypi-sys-mods` is not the yes for install a 3rd party repo. compareable with java license agreement or lets encrypt agreement. if you want to install them automated you have to set an environment variable or provide a specific parameter.

Reply:
As I am not a normal maintainer for this repository, I did not feel that I should change the default behavior. I will leave it up to the true maintainers to modify the default if they accept my PR.

The mechanism I am using here to prompt the user (or not for that matter) is debconf. Debconf is well integrated into the Debian ecosystem and you have probably already used it without knowing. It is the same thing that is used to select your language, keyboard, and timezone among other things. These choices can be preloaded, and persisted across updates/installs (unless the package is purged). 

Under normal circumstances, Debconf will prompt the user if:

- They haven't seen the question before / they haven't already added in the value ahead of time
- The priority of the question is at or above the priority they wish to see (low, medium, high, critical)
- Debconf can interact with their specified front-end

If cases  the user is not prompted, then the default is used.

You can see the choices of frontends you have available by running:
`sudo dpkg-reconfigure -plow debconf`
Your current selections for debconf should already be set at their current value.

So to specify your own default, you would just need to preload the debconf database. For this change you could use a command something like this:
`echo "raspberrypi-sys-mods raspberrypi-sys-mods/allow_3p_repos select no" | sudo debconf-set-selections`

The trick here is that you would have to populate this information prior to installing the packing. However, if you are running interactively, allow prompting of medium priority questions, and do not preconfigure the value you will still be prompted anyway.

Software designed to allow building of custom Raspberry Pi images should already have hooks for debconf. In fact, the offical builder, [pi-gen](https://github.com/RPi-Distro/pi-gen), documents that hook right on the readme (Search for 00-debconf).

I am not aware of any GDPR requirement that this package would be in breach of (although I am not a lawyer). There are certain compliance agencies, such as NERC, that are very strict. In such a case, they would not be using stock images unless they were heavily vetted. All changes would be going through strict change control. 

For general usage, I feel that debconf is a excellent tool to give end users and administrators the ability to modify package behavior. As I said, I believe the upstream software maintainer(s) should choose the default value. This code just makes it easier to provide that choice.

Reply:
@mikewwwperry it is all about changing the default behavior not keeping it in the trojan state :)

as i am from europe i am quite often confronted with GDPR. i am also no lawyer but what i can say (which a lawyer can hopefully confirm) that tracking have to be opt-in and not opt-out (this is where all this terrible cookie banners come from).

some somple GDPR facts for developers:
- having the M$ repo installed means the user pings M$, they get the users ip -> tracking
- tracking must be opt-in
- default has to be `no`

it is not about allowing the user/developer/builder to disable the tracking by opt-out via a env var or configure by preconfiguring debconf. it is a privacy first requirement. so if someone wants to build an image with the ms repo, then they have to actvily enable it with the debconf tool

i fully understand that you are no core developer and i thank you for your time and effort providing this PR without knowing if it even will be merged.
with my comment i just want to make certain that the maintainers may have a real problem with GDPR in the current repos state and maybe also after they merge this if they don't change the default setting.

Reply:
At this point I think that, yes it should use debconf to ask user contentment. But also in ti's own package as t'its quite an optional `raspberrypi-sys[tem]-modI[ification]s`. 

A nice suggested package called something like vendors_repo.deb or non-free_corporate_repos.deb. 

Anyway, thx for submitting something without knowing if they even plan to fix it.

Reply:
This is the right way - a new package.